Library/Formula/gasnet.rb

```diff
@@ -0,0 +1,166 @@
+class Gasnet < Formula
+  desc "Global-Address Space Networking layer and communication primitives"
+  homepage "http://gasnet.lbl.gov"
+  url "http://gasnet.lbl.gov/GASNet-1.26.0.tar.gz"
+  mirror "https://bitbucket.org/berkeleylab/gasnet/downloads/GASNet-1.26.0.tar.gz"
+  sha256 "32b84c47910c36137db198a9c5396dd0a87c29ad72f2040493aa9e8b2a24f929"
+
+  head do
+    url "https://bitbucket.org/berkeleylab/gasnet.git"
+
+    depends_on "autoconf"   => :build
+    depends_on "automake"   => :build
+    depends_on "libtool"    => :build
+    depends_on "pkg-config" => :build
+  end
+
+  option "without-test", "Skip build time tests (requires properly configured MPI and firewall settings)"
+
+  depends_on :mpi => [:cc, :cxx]
+
+  def install
+    system "./bootstrap" if build.head?
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--prefix=#{prefix}"
+    system "make", "all"
+    system "make", "run-tests" if build.with? "test"
+    system "make", "install"
+  end
+
+  test do
+    # This code is blatantly stolen from the package itself, and can be found here:
+    # https://bitbucket.org/berkeleylab/gasnet/src/15ce4a01817fea10c43447e25bbda5bae72f7452/tests/testhello.c?at=master
```

Comment:
Good thinking.... I'm new to the whole installing files via the Formula (as opposed to `make install`) thing. I just fixed this up to a decent state, and pushed the changes.